test/deploy-eks-cluster.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+function deleteCluster() {
+  eksctl delete cluster --name=${CLUSTER_NAME}
+  if [ $? -ne 0 ]; then
+    echo "Unable to delete cluster - ${CLUSTER_NAME}"
+    return 1
+  fi
+
+  return 0
+}
+
+function createCluster() {
+  # Deploy eksctl cluster if not deploy
+  rc=$(which eksctl)
+  if [ -z "$rc" ]; then
+    echo "eksctl is not installed or in the PATH. Installing..."
+    os=$(uname -s)
+    curl --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_${os}_amd64.tar.gz" | tar xz -C /tmp
+    if [ $? -ne 0 ]; then
+     echo "Unable to download eksctl for ${os}."
+     return 1
+    fi
+    chmod +x /tmp/eksctl
+    sudo mv /tmp/eksctl /usr/local/bin
+  fi
+
+  found=$(eksctl get cluster --name "${CLUSTER_NAME}")
+  if [ -z "${found}" ]; then
+    eksctl create cluster --name=${CLUSTER_NAME} --nodes=${NUM_WORKERS}
+    if [ $? -ne 0 ]; then
+      echo "Unable to create cluster - ${CLUSTER_NAME}"
+      return 1
+    fi
+  else
+    echo "Retrieving kubeconfig for ${CLUSTER_NAME}"
+    # Cluster exists but kubeconfig may not
+    eksctl utils write-kubeconfig --cluster=${CLUSTER_NAME}
+  fi
+
+  echo "Logging in to ECR"
+  rc=$(aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 667741767953.dkr.ecr.us-west-2.amazonaws.com/splunk/splunk-operator)
```

Comment:
Account info should probably come from ENV

Reply:
Updated to use PRIVATE_REGISTRY